pyiron/thermodynamics/hessian.py

```diff
@@ -18,40 +18,117 @@
-        }
+        self.server.run_mode.interactive = True
         self._force_constants = None
         self._reference_structure = None
-        self._next_positions = None
+        self._energy_pot = 0
+        self._forces = np.zeros(3)
+        self._pressure = np.zeros((3,3))
+        self._stiffness_tensor = np.zeros((6,6))
+        self._pressure_times_volume = 0
+
+    def set_elastic_moduli(self, bulk_modulus=0, shear_modulus=0):
+        self._stiffness_tensor = np.zeros((6,6))
+        self._stiffness_tensor[:3,:3] = bulk_modulus-2*shear_modulus/3
+        self._stiffness_tensor[:3,:3] += np.eye(3)*2*shear_modulus
+        self._stiffness_tensor[3:,3:] = np.eye(3)*shear_modulus
 
     def set_force_constants(self, force_constants):
-        force_shape = np.shape(force_constants)
-        force_reshape = force_shape[0] * force_shape[2]
-        self._force_constants = np.transpose(force_constants, (0, 2, 1, 3)).reshape((force_reshape, force_reshape))
+        if self.structure is None:
+            raise ValueError('Set reference structure via set_reference_structure() first')
+        n_atom = len(self.structure.positions)
+        if len(np.array([force_constants]).flatten())==1:
+            self._force_constants = force_constants*np.eye(3*n_atom)
+        elif np.array(force_constants).shape==(3*n_atom, 3*n_atom):
+            self._force_constants = force_constants
+        elif np.array(force_constants).shape==(n_atom, n_atom):
+            na = np.newaxis
+            self._force_constants = (np.array(force_constants)[:,na,:,na]*np.eye(3)[na,:,na,:]).flatten()
+        elif len(np.shape(force_constants))==4:
+            force_shape = np.shape(force_constants)
+            if force_shape[2]==3 and force_shape[3]==3:
+                force_reshape = force_shape[0] * force_shape[2]
+                self._force_constants = np.transpose(force_constants, (0, 2, 1, 3)).reshape((force_reshape, force_reshape))
+            elif force_shape[1]==3 and force_shape[3]==3:
+                self._force_constants = np.array(force_constants).reshape(3*n_atom, 3*n_atom)
+            else:
+                raise AssertionError('force constant shape not recognized')
+        else:
+            raise AssertionError('force constant shape not recognized')
 
     def set_reference_structure(self, structure):
-        self._reference_structure = structure
-        self.structure = structure
+        self._reference_structure = structure.copy()
+        if self.structure is None:
+            self.structure = structure.copy()
+
+    def validate_ready_to_run(self):
+        super(HessianJob, self).validate_ready_to_run()
+        if self._force_constants is None:
+            raise AssertionError('set force constants by set_force_constants before run')
+        if self._reference_structure is None:
+            raise AssertionError('set reference structure by set_reference_structure before run')
+
+    def interactive_forces_getter(self):
```

Comment:
I do not think this is necessary if we use the interactive_cache.

Reply:
That's true. Either way is possible. The slight advantage of defining getters is that the user can also remove some of them. But in this job probably this plays no significant role.

Reply:
No from my perspective that is a general design decision to implement the cache for all jobs.